Retrieve node key pair information using admin API

chris-j-h opened the pull request:
* Add `adminhost:port/config/keypairs` endpoint to retrieve all key pairs for a particular node.  Example response is:
```json
[
   {
      "publicKey" : "oNspPPgszVUFw0qmGFfWwh1uxVUXgvBxleXORHj07g8="
   },
   {
      "publicKey" : "ABn6zhBth2qpdrJXp98IvjExV212ALl3j4U//nj4FAI="
   }
]
```
*  Add `adminhost:port/config/keypairs/{publickey}` endpoint to check if node is registered with a particular key pair.  Example response is:
```json
  {
      "publicKey" : "oNspPPgszVUFw0qmGFfWwh1uxVUXgvBxleXORHj07g8="
   }
```
if the node is registered with the key pair, and:
```
No key pair found with public key oNspPPgszVUFw0qmGFfWwh1uxVUXgvBxleXORHj07g8=
```
if node is not registered with the key pair

chris-j-h committed:
Add endpoints to retrieve the key pairs for a node

chris-j-h committed:
Merge remote-tracking branch 'origin/master'

chris-j-h committed:
Merge remote-tracking branch 'origin/master'

chris-j-h committed:
Retrieve keys from enclave for admin api

chris-j-h committed:
Fix typo

chris-j-h committed:
Add notes

prd-fox commented on jaxrs-service/src/main/java/com/quorum/tessera/admin/ConfigResource.java:
```diff
@@ -85,4 +90,49 @@ public Response getPeers() {
+    public Response getKeyPairs() {
+        Set<PublicKey> publicKeys = configService.getPublicKeys();
+
+        if(publicKeys.isEmpty()) {
```
It would be better for return the empty list instead of throwing an error here; the resource exists, but is just empty.

prd-fox commented on jaxrs-service/src/main/java/com/quorum/tessera/admin/ConfigResource.java:
```diff
@@ -85,4 +90,49 @@ public Response getPeers() {
+            .map(publicKey -> {
+                KeyData kd = new KeyData();
+                kd.setPublicKey(publicKey);
+                kd.setPrivateKey("REDACTED");
```
How about creating a new object that only contains the public key? The "REDACTED" and type in the response isn't very useful for the user.

chris-j-h replied:
This was something I had considered doing - I'll add it in

chris-j-h committed:
Add dedicated transport object & return empty list if no data found
New DTO improves formatting of response data

chris-j-h committed:
Ensure immutability of PublicKeyResponse
Remove setter and replace with constructor, make default constructor private so it cannot be used but is present for marshalling
